tests/system/python/packages/test_available_and_install_api.py

```diff
@@ -153,32 +169,52 @@ def test_install_plugin_package(self, foglamp_url, package_build_source_list, pa
                     for pkg_name in pkg_list_name:
                         full_pkg_name = 'foglamp-{}-{}'.format(k, pkg_name)
                         if full_pkg_name in available_pkg:
-                            self._verify_and_install_package(foglamp_url, full_pkg_name)
+                            self._verify_and_install_package(foglamp_url, full_pkg_name, fp)
                         else:
                             # TODO: Add them into some csv (report)
                             print("{} not found in available package list".format(full_pkg_name))
 
-    def _verify_and_install_package(self, foglamp_url, pkg_name):
+        conn = http.client.HTTPConnection(foglamp_url)
+        conn.request("GET", '/foglamp/plugins/installed')
+        r = conn.getresponse()
+        assert 200 == r.status
+        r = r.read().decode()
+        jdoc = json.loads(r)
+        assert len(jdoc), "No data found"
+        assert counter == len(jdoc['plugins'])
+
+        fp.close()
+
+    def _verify_and_install_package(self, foglamp_url, pkg_name, fp):
         global counter
+        global plugin_count
         conn = http.client.HTTPConnection(foglamp_url)
         data = {"format": "repository", "name": pkg_name}
         conn.request("POST", '/foglamp/plugins', json.dumps(data))
         r = conn.getresponse()
         assert 200 == r.status
         r = r.read().decode()
         jdoc = json.loads(r)
-        assert '{} is successfully installed'.format(pkg_name) == jdoc['message']
+        if '{} is successfully installed'.format(pkg_name) == jdoc['message']:
```

Comment:
CSV report status on the basis of HTTP_STATUS_CODE; if 200 then Passed otherwise Failed. And right place of this check is before HTTP status code assertion